tests/Doctrine/Tests/ORM/Functional/Ticket/Issue7052Test.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Tests\Models\CMS\CmsEmployee;
+
+/**
+ * @group issue-7052
+ */
+class Issue7052Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(array(
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\Issue7052Child'),
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\Issue7052Parent'),
+        ));
+    }
+
+    public function testIssue()
+    {
+        $parent = new Issue7052Parent();
+        $parent->name = "parent test";
+
+        $childA = new Issue7052Child();
+        $childA->name = "child A";
+        $childA->parent = $parent;
+
+        $childB = new Issue7052Child();
+        $childB->name = "child B";
+        $childB->parent = $parent;
+
+        $this->_em->persist($parent);
+        $this->_em->persist($childA);
+        $this->_em->persist($childB);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $childA = $this->_em->find('Doctrine\Tests\ORM\Functional\Ticket\Issue7052Child', $childA->id);
+
+        $this->_em->clear();
+
+        $childB = $this->_em->find('Doctrine\Tests\ORM\Functional\Ticket\Issue7052Child', $childB->id);
+
+        $parentFromChildB = $childB->parent;
+
+        $this->assertNotNull($childA->parent->name, "Unable to get parent name after EM cleared.");
```

Comment:
Are the identifiers set? Is the issue only around the state of `$name`?

Reply:
Yes, the identifier is set (which is pretty strange as we go through this if: https://github.com/doctrine/doctrine2/blob/2.5/lib/Doctrine/ORM/UnitOfWork.php#L2531 ), but all others properties are at null. Here only the name, because it's the only property apart from the id.
But I can add other properties and test them if that makes sense. That would maybe make the code a little bit less straight to the point.